src/Components/CreateImageWizard/validators.ts

```diff
@@ -79,6 +84,27 @@ export const isSshKeyValid = (sshKey: string) => {
   return isPatternValid;
 };
 
+export const isPasswordValid = (password: string) => {
+  if (!password) {
+    return true;
+  }
+  const isEncrypted = /^\$(6|5|2b)\$/.test(password);
+  const isLengthValid = password.length >= 8 && password.length <= 128;
+  const hasUpperCase = /[A-Z]/.test(password);
+  const hasLowerCase = /[a-z]/.test(password);
+  const hasSpecialChar = /[^A-Za-z0-9]/.test(password);
+  const hasNoSpaces = !/\s/.test(password);
+
+  if (isEncrypted) return true;
+  return (
+    isLengthValid &&
+    hasUpperCase &&
+    hasNoSpaces &&
+    hasLowerCase &&
+    hasSpecialChar
+  );
+};
```

Comment:
I might be misunderstanding something but as far as I can tell in the image definitions only Azure images seem to have any restrictions.

https://github.com/search?q=repo%3Aosbuild/images%20PwqualityConfConfig&type=code

If you haven't done so, you can always test this by building and booting an image with a simple password.

Remember, the goal of our validation is to prevent users from accidentally building images that fail to build or boot but to otherwise be as unopinionated / not-annoying as possible. And password rules are definitely annoying. :wink:

Reply:
ok, I did some tests in rhel env, and the only limit for password is that the minimum length is 6.
so I changed it to 6.
now we need to decide if we want to add warning to the user if he fill weak password - 
like only small letters or only digit?
and then we can put a link how to create strong password 
https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/4/html/security_guide/s1-wstation-pass#s2-wstation-pass-create
wdyt?